### PR TITLE
Compare string attributes

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Add helpers to compare string attributes + use them for validating range statements.

## Changes

_Describe the changes that were needed.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/common/src/web3IdHelpers.ts [hunk 1]
```diff
@@ -21,3 +21,29 @@ export function verifyWeb3IdCredentialSignature(
...
+ * if a > b then compareStringAttributes(a,b) = 1;
+ */
+export function compareStringAttributes(a: string, b: string): number {
+    return wasm.compareStringAttributes(a, b);
+}
+
+/**
+ * Given a string attribute value and a range [lower, upper[, return whether value is in the range, when converted into field elements.
+ */
+export function isStringAttributeInRange(
```
Comment (orhoj) [inline packages/common/src/web3IdHelpers.ts:38]: Make unit tests for the edge cases where `value === lower` and `value === upper`.

File: packages/common/src/web3Proofs.ts [hunk 2]
```diff
@@ -103,7 +109,7 @@ function verifyRangeStatement(
             }
         };
 
-        if (properties.type === 'string' && properties.format === 'date-time') {
+        if (isTimeStamp(properties)) {
```
Comment (orhoj) [inline packages/common/src/web3Proofs.ts:112]: I am unsure if we actually agreed to parse it is a date when `format = date-time` or if we decided to have a custom `type = timestamp` for this.

**Review by orhoj — CHANGES_REQUESTED** [review]

File: packages/common/src/web3Proofs.ts (hunk 2 above)
Comment (shjortConcordium) [reply packages/common/src/web3Proofs.ts]: Fixed this.

File: packages/common/src/web3IdHelpers.ts (hunk 1 above)
Comment (shjortConcordium) [reply packages/common/src/web3IdHelpers.ts:38]: Added these.
What would we expect if `value === lower === upper`? :thinking:

Comment (orhoj) [reply packages/common/src/web3IdHelpers.ts:38]: That is invalid input as it would result in an empty interval as `[a, a) = ∅`.

**Review by orhoj — CHANGES_REQUESTED** [review]

File: packages/common/src/web3ProofTypes.ts [hunk 3]
```diff
@@ -39,15 +39,32 @@ export type Web3IdProofInput = {
...
+    description?: string;
+    type: 'string' | 'integer';
     format?: string;
 };
 
+export type PropertyDetails = SimplePropertyDetails | TimestampPropertyDetails;
+
 type IndexDetails = {
-    title: 'id';
+    title: string;
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/common/src/web3ProofTypes.ts:67]: I don't think the title needs to be id, at least it is not in our example :thinking:  
https://github.com/Concordium/concordium-web3id/blob/main/examples/json-schemas/education-certificate/JsonSchema2023-education-certificate.json

**Review by orhoj — APPROVED** [review]
